### PR TITLE
RavenDB-7727 making project compile in Release

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs [hunk 1]
```diff
@@ -208,13 +208,11 @@ private void ReplicateToDestination()
                         {
                             while (true)
                             {
-#if DEBUG
```
Comment (ayende) [inline src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs:211]: This needs to be wrapped in a conditional method, not removed.

Comment (ppekrol) [reply src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs:211]: It can't, we would have to remove tests that are using this.

Comment (ayende) [reply src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs:211]: `[Conditional("DEBUG")]` will allow it to be compiled in Release, but not actually be called.

Comment (fitzchak) [reply src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs:211]: @ppekrol Are we running all of our tests in Debug mode, right?
As we have lots of Assert.Debug which should be tested.
If so, can we skip compiling the tests in Release mode?

Comment (ppekrol) [reply src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs:211]: We don't, we are always compiling and running in Release. All versions, 2.5/3.0/3.5/4.0.

Comment (fitzchak) [reply src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs:211]: So I think that we need to run the tests of v4.0 in Debug mode, as we have lots of Debug.Assert in the code base and that wasn't the case in v3.5 and before.

Comment (ayende) [reply src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs:211]: I'm always running them in `Debug` mode locally

**Pull request merged**